core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -512,15 +514,12 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, long offset,
         openMs = System.currentTimeMillis() - startMs;
         blockReader = context.getBlockReader();
         Preconditions.checkState(blockReader != null);
-        ByteBuf buf = PooledByteBufAllocator.DEFAULT.buffer(len, len);
         try {
           long startTransferMs = System.currentTimeMillis();
-          while (buf.writableBytes() > 0 && blockReader.transferTo(buf) != -1) {
```

Comment:
Interesting, I thought the transferTo + pooledBuf would be more efficient.   Have you run any benchmark?